sktime/transformations/series/summarize.py

```diff
@@ -3,13 +3,446 @@
 # copyright: sktime developers, BSD-3-Clause License (see LICENSE file)
 """Implement transformers for summarizing a time series."""
 
-__author__ = ["mloning", "RNKuhns"]
-__all__ = ["SummaryTransformer"]
+__author__ = ["mloning", "RNKuhns", "danbartl"]
+__all__ = ["SummaryTransformer", "WindowSummarizer"]
 
 import pandas as pd
+from joblib import Parallel, delayed
 
 from sktime.transformations.base import BaseTransformer
 
+
+class WindowSummarizer(BaseTransformer):
+    """Transformer for extracting time series features.
+
+    The WindowSummarizer transforms input series to features based
+    on a provided dictionary of window summarizer, window shifts
+    and window lengths.
+
+    Parameters
+    ----------
+    n_jobs : int, optional (default=-1)
+        The number of jobs to run in parallel for applying the window functions.
+        ``-1`` means using all processors.
+    target_cols: list of str, optional (default = None)
+        Specifies which columns in X to target for applying the window functions.
+        ``None`` will target the first column
+    lag_config: dict of str and list, optional (default = dict containing first lag)
+        Dictionary specifying as key the `name` of the columns to be
+        generated. As value the dict specifies the type of function via the argument
+        `summarize` as well as the length 2 list argument `window`. The list `window`
+        will be resolved by the internal function _window_feature to `window length`
+        - the length of the window across which to apply the function - as well as
+        the argument `starting_at`, which will specify how far back in the past
+        the window will start.
+
+        For example for `window = [4, 3]`, we have a `window_length` of 4 and
+        `starting_at` of 3 to target the four days prior to the last three days.
+        Here is a representation of the selected window:
+
+        |-------------------------------|
```

Comment:
@fkiraly , I think the documentation here is still not consistent with #2290. For example (assuming `z` is the current point and `x` are the prior points , then per #2290, for a window of [4, 3], we should get the following selected window (indicated by `y`)

| x x x x x x x x x x x x x x x x x x z - - - |
| x x x x x x x x x x x x x x x x y y y - - - |

Reply:
yes, you are right, @ngupta23.
I misread this, thinking that xxxz is the window?
The docstring is wrong, so I hope the implementation isn't...

The below doesn't make much sense, the window in this example has lenght 7.